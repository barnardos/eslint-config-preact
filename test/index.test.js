const eslint = require('eslint');
const path = require('path');
const fs = require('fs');

const cli = new eslint.CLIEngine({
	fix: false,
	cache: false,
	errorOnUnmatchedPattern: true
});
const formatter = cli.getFormatter('tap');
function lint (file) {
	const report = cli.executeOnFiles(file);
	const text = formatter(report.results).replace(/^TAP.+\n.+\n.+\n/g, '');
	return Object.assign({ text }, report);
}

const FIXTURES = path.resolve(__dirname, 'fixtures');
fs.readdirSync(FIXTURES).forEach(dir => {
	if (dir[0] === '.') return;

	const p = f => path.resolve(FIXTURES, dir, f);

	describe(`Fixture: ${dir}`, () => {
		test('valid', () => {
			const report = lint(p('valid.js'));
			expect(report.text).not.toBeTruthy();
			expect(report).toHaveProperty('errorCount', 0);
			expect(report).toHaveProperty('warningCount', 0);
			expect(report.text).toMatchSnapshot(`fixtures/${dir}/valid.js`);
		});
		test('invalid', () => {
			const report = lint(p('invalid.js'));
			expect(report.text).toBeTruthy();
			expect(report.errorCount + report.warningCount).toBeGreaterThan(0);
			expect(report.text).toMatchSnapshot(`fixtures/${dir}/invalid.js`);
		});
	});
});
