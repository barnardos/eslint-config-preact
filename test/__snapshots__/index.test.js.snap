// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixture: component invalid: fixtures/component/invalid.js 1`] = `
"  ---
  message: Component should be written as a pure function
  severity: warning
  data:
    line: 3
    column: 8
    ruleId: react/prefer-stateless-function
  messages:
    - message: Useless constructor.
      severity: warning
      data:
        line: 4
        column: 2
        ruleId: no-useless-constructor
  ...
"
`;

exports[`Fixture: component valid: fixtures/component/valid.js 1`] = `""`;
